examples/hybrid/staggered_nonhydrostatic_model.jl

```diff
@@ -166,6 +166,9 @@ function implicit_tendency_special!(Yₜ, Y, p, t)
 
     ref_thermo_params = Ref(thermo_params)
     ref_zuₕ = Ref(zero(eltype(Yₜ.c.uₕ)))
+    FT = eltype(Yₜ)
+    parent(Yₜ.c) .= FT(0)
+    parent(Yₜ.f) .= FT(0)
```

Comment:
We should probably add this to the generic one, too. Also, can we move this into the threaded loop in this function so that we leverage threading?

Reply:
>  Also, can we move this into the threaded loop in this function so that we leverage threading?
can you suggest a code?

Reply:
Do you mean to do it in line 175 like this:

` @. Yₜ.c[colidx] = FT(0)
`